src/color.rs

```diff
@@ -350,6 +350,168 @@ macro_rules! impl_lch_like {
 impl_lch_like!(Lch, "lch");
 impl_lch_like!(Oklch, "oklch");
 
+/// A Predefined color space specified in:
+/// https://w3c.github.io/csswg-drafts/css-color-4/#predefined
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum PredefinedColorSpace {
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-sRGB
+    Srgb,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-sRGB-linear
+    SrgbLinear,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-display-p3
+    DisplayP3,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-a98-rgb
+    A98Rgb,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-prophoto-rgb
+    ProphotoRgb,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-rec2020
+    Rec2020,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-xyz
+    XyzD50,
+    /// https://w3c.github.io/csswg-drafts/css-color-4/#predefined-xyz
+    XyzD65,
+}
+
+impl PredefinedColorSpace {
+    /// Returns the string value of the predefined color space.
+    pub fn as_str(&self) -> &str {
+        match self {
+            PredefinedColorSpace::Srgb => "srgb",
+            PredefinedColorSpace::SrgbLinear => "srgb-linear",
+            PredefinedColorSpace::DisplayP3 => "display-p3",
+            PredefinedColorSpace::A98Rgb => "a98-rgb",
+            PredefinedColorSpace::ProphotoRgb => "prophoto-rgb",
+            PredefinedColorSpace::Rec2020 => "rec2020",
+            PredefinedColorSpace::XyzD50 => "xyz-d50",
+            PredefinedColorSpace::XyzD65 => "xyz-d65",
+        }
+    }
+}
+
+impl std::str::FromStr for PredefinedColorSpace {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s.to_lowercase().as_str() {
+            "srgb" => PredefinedColorSpace::Srgb,
+            "srgb-linear" => PredefinedColorSpace::SrgbLinear,
+            "display-p3" => PredefinedColorSpace::DisplayP3,
+            "a98-rgb" => PredefinedColorSpace::A98Rgb,
+            "prophoto-rgb" => PredefinedColorSpace::ProphotoRgb,
+            "rec2020" => PredefinedColorSpace::Rec2020,
+            "xyz-d50" => PredefinedColorSpace::XyzD50,
+            "xyz-d65" => PredefinedColorSpace::XyzD65,
+
+            _ => return Err(()),
+        })
+    }
+}
+
+impl ToCss for PredefinedColorSpace {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result
+    where
+        W: fmt::Write,
+    {
+        dest.write_str(self.as_str())
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for PredefinedColorSpace {
```

Comment:
I don't understand, can't this be derived? Serializing as a string looks rather inefficient

Reply:
If I derive on `PredefinedColorSpace` then I have to add `#[cfg_attr(feature = serde, serde(rename = "???"))]` to every item in the enum.  I still want the `FromStr`, but I don't want to `serde::deserialize` there, i just want a match.  I don't want to maintain 2 lists of constants.

If you say serialize to string is inefficient, do you mean serialize to something else?

Reply:
Also, I'm not even sure we need serialization for the color struct any more as we're matching against css for testing now. But another issue I guess.

Reply:
I'm confused, why do we need the serialization anymore? It was useful for tests iirc.

Now we don't need it, and we don't allow serializing the whole `Color` struct anyways. So I'd rather get rid of it.

Specially since serializing from / to string is rather inefficient (color space should just be serialized as an integer for the discriminant).

Reply:
Removed serialization where we don't use it in tests.